registry/storage/driver/s3-aws/s3.go

```diff
@@ -1187,3 +1213,39 @@ func (w *writer) flushPart() error {
 	w.pendingPart = nil
 	return nil
 }
+
+func createHTTPClient(userAgent, rootCA string, skipVerify bool) *http.Client {
+	if userAgent == "" && rootCA == "" && skipVerify == false {
+		return nil
+	}
+
+	httpTransport := http.DefaultTransport
+	var tlsConfig *tls.Config
+
+	if skipVerify || rootCA != "" {
+		tlsConfig = &tls.Config{}
+		if skipVerify {
+			tlsConfig.InsecureSkipVerify = true
+		}
+		if rootCA != "" {
+			caCertPool := x509.NewCertPool()
+			caCertPool.AppendCertsFromPEM([]byte(rootCA))
+			tlsConfig.RootCAs = caCertPool
+		}
+
+	}
+
+	if tlsConfig != nil {
+		httpTransport = &http.Transport{TLSClientConfig: tlsConfig}
```

Comment:
nit: can simplify by putting in the above `if` and getting ride of the outer-scoped `tlsConfig`